cpp/CMakeLists.txt

```diff
@@ -750,6 +750,11 @@ if(ARROW_JEMALLOC)
   list(APPEND ARROW_SYSTEM_LINK_LIBS jemalloc::jemalloc)
 endif(ARROW_JEMALLOC)
 
+if(ARROW_MIMALLOC)
+  add_definitions(-DARROW_MIMALLOC)
+  list(APPEND ARROW_SYSTEM_LINK_LIBS mimalloc::mimalloc)
+endif(ARROW_MIMALLOC)
```

Comment:
Could you remove `ARROW_MIMALLOC` here?
Normally, we don't put any content to `endif()`.

(We can also remove `ARROW_JEMALLOC` in `endif(ARROW_JEMALLOC)`.)